FestivalPlanner/films/views.py

```diff
@@ -309,6 +299,31 @@ def render_to_response(self, context, **response_kwargs):
         BaseFilmsFormView.found_films = None
         return response
 
+    def _filter_films(self, session):
+        pr_debug('start filtered query', with_time=True)
+        filter_kwargs = {'festival': self.festival}
+        if self.shorts_filter.on(session):
+            filter_kwargs['duration__gt'] = self.short_threshold
+        self.selected_films = Film.films.filter(**filter_kwargs).order_by('seq_nr')
+        for fan in self.fan_list:
+            if self.rated_filters[fan].on(session):
+                self.selected_films = self.selected_films.filter(
+                    ~Exists(FilmFanFilmRating.film_ratings.filter(film=OuterRef('pk'), film_fan=fan))
+                )
```

Comment:
Good thing to split this off in a protected method.

Please consider to group protected and unprotected methods together, and see if more class methods can be made protected.

Reply:
Done.